cmd/test-server/kcp/shard.go

```diff
@@ -114,26 +133,28 @@ func Start(ctx context.Context, name, runtimeDir, logFilePath string, args []str
 		}
 	}()
 
+	// Start a goroutine that will notify when the process has exited
 	terminatedCh := make(chan error, 1)
+	s.terminatedCh = terminatedCh
 	go func() {
 		terminatedCh <- cmd.Wait()
```

Comment:
then we could use `s.terminatedCh` here

Reply:
I don't think this will work, because `s.terminatedCh` is a receive-only channel, so we need to use the unconstrained version here